### ISSUE TITLE
Ordinal Special Exceptions (Context/Plurality)

### ISSUE THREAD

**Issue by thunsaker** [Humanizr/Humanizer#212]
This came about from a conversation on #188 [comment](https://github.com/MehdiK/Humanizer/pull/188#issuecomment-40256089). Hopefully this make sense to everyone. 
### Context

I propose adding a "context" enum for special cases where (in Spanish) the spelling can be affected by the part of speech that accompanies it. Something like an override for `number.ToOrdinalWords(GrammaticalGender gender, GrammaticalContext context)`

Maybe something like:

```
public enum GrammaticalContext {
    Noun,
    Adjective,
    ...
    NotApplicable
}
```

**Dates**
Do we need a way to handle humanizing dates with ordinals? ie: Converting April 4, 2014 to April Fourth Two Thousand Fourteen, which can already be done with existing API methods, but in pieces, far from convenient.

Using Spanish as an example, we would have a problem with Humanizing a date, since only the first (el primer de marzo) of the month uses the ordinal, the rest are just the cardinal number (el cinco de mayo). Are there examples in other languages or is this something that a developer would just have to know about?

**Part of Speech**
Depending on the part of speech whether we are talking about a noun, the 1st book (el primer libro) the "o" in primero gets dropped only in the masculine form, however it is preceded by anything else, the 1st to leave (el primero en salir) the "o" is retained. This also happens with tercero (3rd).
### Plurality

If we are translating the ordinal into a plural form, the 1st to arrive (los primeros en llegar), the "o" is restored along with the added "s"

Maybe this could be addressed by having an overridable `number.ToOrdinalWords(GrammaticalGender gender, Plurality plurality)` method added to the exiting interface. Or just add one to Spanish if not needed in other languages.

**tl;dr - Need to handle special rule when 1st/3rd are preceded by a singular masculine noun**
